### PR TITLE
Include S/N ratio for extracted spectra in the ``SpecObj`` datamodel

### PR DESCRIPTION
**Pull request by tbowers7**
The median signal-to-noise ratios for extracted spectra are used in various places (flexure, collation) and also computed for the ``spec1d`` text output table, but are not explicitly included in the ``SpecObj`` data model (and hence included in the ``spec1d`` FITS headers).

This PR adds the computed S/N ratio to that data model and also modifies the users of the value to point to the datamodel value rather than recompute it.

### CODE REVIEW

**Review by kbwestfall — APPROVED** [review]
Nice!  No notes.

**Review by profxj — APPROVED** [review]
Good stuff, thx!

**Comment by tbowers7** [comment]
Before I merge this, I am curious if anyone knows if there was a particular reason the `S2N` was left out of the `SpecObj` datamodel in the first place.  Given some of the things I had to change, it almost seemed purposeful.

File: pypeit/scripts/collate_1d.py
Comment (tbowers7) [inline pypeit/scripts/collate_1d.py]: Perhaps for @kbwestfall : Since this PR moves the S/N value into a datamodel entry, everything in `spec_obj` is now an attribute without any properties.  Is there a better way to produce the `spec_obj_data` list in this case?

**Comment by profxj** [comment]
If there was a reason, it has long since escaped my memory..

File: pypeit/scripts/collate_1d.py
Comment (kbwestfall) [reply pypeit/scripts/collate_1d.py]: This approach is still fine.  Because of the way the `DataContainer` class is constructed, this approach is almost identical to, e.g., `[spec_obj[x] for x in spec_obj_keys]`.

Comment (kbwestfall) [reply pypeit/scripts/collate_1d.py]: I think you can remove the comment.

**Pull request merged**
